compiler/src/pipeline.rs

```diff
@@ -247,6 +266,26 @@ impl<T: FieldElement> Pipeline<T> {
         }
     }
 
+    pub fn with_setup_file(self, setup_file: Option<PathBuf>) -> Self {
+        Pipeline {
+            arguments: Arguments {
+                setup_file,
+                ..self.arguments
+            },
+            ..self
+        }
+    }
+
+    pub fn with_existing_proof_file(self, existing_proof_file: Option<PathBuf>) -> Self {
+        Pipeline {
+            arguments: Arguments {
+                existing_proof_file,
+                ..self.arguments
+            },
+            ..self
+        }
+    }
```

Comment:
this can be clearer and more succinct

```suggestion
    pub fn with_existing_proof_file(mut self, existing_proof_file: Option<PathBuf>) -> Self {
        self.arguments.existing_proof_file = existing_proof_file;
        self
    }
```